File: techila/examples/R/Features/custom_library/packagetest_dist.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This script contains the Worker Code, which contains the
# packagetest_dist function that will be executed in each
# computational Job.

# Load the The techilaTestPackage library. 
library(techilaTestPackage)

packagetest_dist <- function(input) {
  # Call the techilaTestFunction from the techilaTestPackage
  result <- techilaTestFunction(input) 
}

File: techila/examples/python/Features/custom_package/packagetest_dist.py
# Copyright 2012-2013 Techila Technologies Ltd.

# This script contains the Worker Code, which contains the
# packagetest_dist function that will be executed in each
# computational Job.

# Load the the techilatest module.
import techilatest.functions as test

# Define the function that will be called on Workers
def packagetest_dist(a, b):
    # Call the functions defined in the 'techilatest' package
    res1 = test.summation(a,b)
    res2 = test.multiplication(a,b)
    res3 = test.subtraction(a,b)
    res4 = test.division(a,b)

    # Return results in a list
    return([res1, res2, res3, res4])
